server/graphql/v2/mutation/RootMutations.ts

```diff
@@ -49,6 +49,61 @@ const BanAccountResponse = new GraphQLObjectType({
  * Root mutations
  */
 export default {
+  editAccountFlags: {
+    type: new GraphQLNonNull(Account),
+    description: '[Root only] Edits account flags (deleted, banned, archived, trusted host)',
+    args: {
+      account: {
+        type: new GraphQLNonNull(AccountReferenceInput),
+        description: 'Account to change the flags for',
+      },
+      isDeleted: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is soft deleted',
+      },
+      isBanned: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is banned',
+      },
+      isArchived: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is archived',
+      },
+      isTrustedHost: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is a trusted host',
+      },
+    },
+    async resolve(_: void, args, req: express.Request): Promise<Record<string, unknown>> {
+      checkRemoteUserCanRoot(req);
+      const account = await fetchAccountWithReference(args.account, { throwIfMissing: true, paranoid: false });
+
+      if (args.isDeleted && !account.deletedAt) {
+        await account.destroy();
+      } else if (!args.isDeleted && account.deletedAt) {
+        await account.restore();
+
+        // We have freed the slug, so we have to restore it to its original value
+        const timestampPart = account.slug.match(/-\d+$/)[0];
+        await account.update({ slug: account.slug.replace(timestampPart, '') });
+      }
+
+      if (args.isArchived && !account.deactivatedAt) {
+        await account.update({ deactivatedAt: Date.now() });
+      } else if (!args.isArchived && account.deactivatedAt) {
+        await account.update({ deactivatedAt: null });
+      }
```

Comment:
:no_entry_sign: Same as delete, we have a process to (un)archive accounts that take care of stuff like making sure we empty the balance first. Let's not bypass that.

Reply:
Thanks, I've used the exiting functions so that this is taken care of. 👍🏽